flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -381,7 +383,7 @@ private void registerAlignmentTimer(
                 registerTimer.registerTask(
                         () -> {
                             try {
-                                operatorChain.alignedBarrierTimeout(checkpointId);
+                                operatorChain.alignedBarrierTimeout(checkpointId, metrics);
```

Comment:
Maybe instead of adding a dependency to `CheckpointMetrics` to all of the call stack down to the subpartition, can `alignedBarrierTimeout` return `true` or `false` depending if the barrier has timed out or not? 🤔

Reply:
Hi @pnowojski , thanks for your review.

It's a great suggestion, updated.